src/bin/rustfmt.rs

```diff
@@ -67,19 +65,17 @@ fn lookup_project_file(dir: &Path) -> io::Result<Option<PathBuf>> {
     loop {
         let config_file = current.join("rustfmt.toml");
         match fs::metadata(&config_file) {
-            Ok(md) => {
-                // Properly handle unlikely situation of a directory named `rustfmt.toml`.
-                if md.is_file() {
-                    return Ok(Some(config_file));
-                }
-            }
-            // If it's not found, we continue searching; otherwise something went wrong and we
-            // return the error.
+            // Only return if it's a file to handle the unlikely situation of a directory named
+            // `rustfmt.toml`.
+            Ok(ref md) if md.is_file() => return Ok(Some(config_file)),
+            // Return the error if it's something other than `NotFound`; otherwise we didn't find
+            // the project file yet, and continue searching.
             Err(e) => {
                 if e.kind() != ErrorKind::NotFound {
```

Comment:
> I think you can legitimately get `ErrorKind::PermissionDenied` here.

I think that erroring out makes more sense. It is more likley that they've borked their permissions than they actually *want* an unreadable file named `rustfmt.toml` to be in the tree.

> Maybe it makes sense to return `Ok(None)` for an unknown error?

I don't think so. If somone has a project `rustfmt.toml` file and there's a filesystem hiccup, I don't think they'd expect or want their project reformatted according to the formating defaults.